erpnext/hr/doctype/attendance/attendance.py

```diff
@@ -78,6 +101,67 @@ def validate_employee(self):
 		if not emp:
 			frappe.throw(_("Employee {0} is not active or does not exist").format(self.employee))
 
+	def calculate_overtime_duration(self):
+		#this method is only for Calculation of overtime based on Attendance through Employee Checkins
+		self.overtime_duration = None
+
+		if not self.standard_working_time and self.shift:
+			self.standard_working_time = frappe.db.get_value("Shift Type", self.shift, "standard_working_time")
+
+		if not self.overtime_type:
+			self.overtime_type = get_overtime_type(self.employee)
+
+		if int(self.working_time) > int(self.standard_working_time):
+			self.overtime_duration = int(self.working_time) - int(self.standard_working_time)
+
+@frappe.whitelist()
+def get_shift_type(employee, attendance_date):
+	emp_shift = frappe.db.get_value("Employee", employee, "default_shift")
+
+	shift_assignment = frappe.db.sql('''SELECT name, shift_type
+		FROM
+			`tabShift Assignment`
+		WHERE
+			docstatus = 1
+			AND employee = %(employee)s AND start_date <= %(attendance_date)s
+			AND (end_date >= %(attendance_date)s OR end_date IS null)
+			AND status = "Active"
+		''', {
+			"employee": employee,
+			"attendance_date": attendance_date,
+		}, as_dict = 1)
+
+	if len(shift_assignment):
+		shift = shift_assignment[0].shift_type
+	else:
+		shift = emp_shift
```

Comment:
```suggestion
		shift = frappe.db.get_value("Employee", employee, "default_shift")
```

Remove the first line for `emp_shift` and compute this directly

Reply:
done